bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1243,73 +1286,223 @@ public void sendNotificationToMobileApp(String customerId, String text, @Nullabl
+            if (announcementQueueRunning.compareAndSet(false, true)) {
+                queuedAnnouncement();
+            }
         }
-        content.speak.value = speak;
+        announcements.clear();
+    }
+
+    public void queuedAnnouncement() {
+        JsonAnnouncement jsonAnnouncement = announcementQueue.poll();
+        if (jsonAnnouncement != null) {
+            String speak = jsonAnnouncement.speak;
+            try {
+                List<Device> devices = jsonAnnouncement.devices;
+                String bodyText = jsonAnnouncement.bodyText;
+                String title = jsonAnnouncement.title;
+                List<Integer> ttsVolumes = jsonAnnouncement.ttsVolumes;
+                List<Integer> standardVolumes = jsonAnnouncement.standardVolumes;
+
+                Map<String, Object> parameters = new HashMap<>();
+                parameters.put("expireAfter", "PT5S");
+                JsonAnnouncementContent[] contentArray = new JsonAnnouncementContent[1];
+                JsonAnnouncementContent content = new JsonAnnouncementContent();
+                if (StringUtils.isEmpty(title)) {
+                    content.display.title = "OpenHAB";
+                } else {
+                    content.display.title = title;
+                }
+                content.display.body = bodyText;
+                content.display.body = speak.replaceAll("<.+?>", "");
+                if (speak.startsWith("<speak>") && speak.endsWith("</speak>")) {
+                    content.speak.type = "ssml";
+                }
+                content.speak.value = speak;
 
-        contentArray[0] = content;
+                contentArray[0] = content;
 
-        parameters.put("content", contentArray);
+                parameters.put("content", contentArray);
 
-        JsonAnnouncementTarget target = new JsonAnnouncementTarget();
-        target.customerId = device.deviceOwnerCustomerId;
-        TargetDevice[] devices = new TargetDevice[1];
-        TargetDevice deviceTarget = new TargetDevice();
-        deviceTarget.deviceSerialNumber = device.serialNumber;
-        deviceTarget.deviceTypeId = device.deviceType;
-        devices[0] = deviceTarget;
-        target.devices = devices;
-        parameters.put("target", target);
+                JsonAnnouncementTarget target = new JsonAnnouncementTarget();
+                target.customerId = devices.get(0).deviceOwnerCustomerId;
+                TargetDevice[] targetDevices = new TargetDevice[devices.size()];
+                for (int i = 0; i < devices.size(); i++) {
+                    Device device = devices.get(i);
+                    TargetDevice deviceTarget = new TargetDevice();
+                    deviceTarget.deviceSerialNumber = device.serialNumber;
+                    deviceTarget.deviceTypeId = device.deviceType;
+                    targetDevices[i] = deviceTarget;
+                }
+                target.devices = targetDevices;
```

Comment:
add a constructor 

```
        public TargetDevice(JsonDevices.Device device) {
            deviceSerialNumber= device.serialNumber;
            deviceTypeId = device.deviceType;
        }
```  

to `JsonAccouncementTarget.TargetDevice` and change this part to

```suggestion
                TargetDevice[] targetDevices = devices.stream().map(TargetDevice::new).collect(Collectors.toList()).toArray(TargetDevice[]::new);
```

Reply:
I have to check this, will provide a solution tomorrow.

Reply:
It seems this doesn't work, got a compilation error.

Reply:
My fault. This is a Java11 feature. Leave it as-is then.